### PR TITLE
DPCPP kNN Regression

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

File: cpp/oneapi/dal/algo/knn/common.hpp [hunk 1]
```diff
@@ -205,7 +230,8 @@ namespace v1 {
 /// @tparam Method      Tag-type that specifies an implementation of algorithm. Can
 ///                     be :expr:`method::brute_force` or :expr:`method::kd_tree`.
 /// @tparam Task        Tag-type that specifies type of the problem to solve. Can
-///                     be :expr:`task::classification`.
+///                     be :expr:`task::classification`, :expr:`task::regression`
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/knn/common.hpp:233]:
```suggestion
///                     be :expr:`task::classification`, :expr:`task::regression`,
```

File: cpp/oneapi/dal/algo/knn/common.hpp [hunk 2]
```diff
@@ -205,7 +230,8 @@ namespace v1 {
...
+///                     or :expr:`task::search`,.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/knn/common.hpp:234]:
```suggestion
///                     or :expr:`task::search`.
```

**Comment by KulikovNikita** [comment]
Last CI: http://intel-ci.intel.com/ec1bc984-6105-f161-8052-8cdcd4b723a1

File: cpp/oneapi/dal/backend/primitives/regression/BUILD [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private = True,
+    srcs = glob([
+        "test/*_dpc.cpp",
+    ], exclude=[
+        "test/*perf*.cpp",
+    ]),
+    dal_deps = [
+        ":regression",
+    ],
+)
```
Comment (KalyanovD) [inline cpp/oneapi/dal/backend/primitives/regression/BUILD:42]: new line

Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/regression/BUILD:42]: will be fixed

**Review by amgrigoriev — APPROVED** [review]

File: cpp/oneapi/dal/algo/knn/backend/cpu/train_kernel_brute_force.cpp [hunk 4]
```diff
@@ -42,6 +44,11 @@ static train_result<Task> call_daal_kernel(const context_cpu& ctx,
                                            const detail::descriptor_base<Task>& desc,
                                            const table& data,
                                            const table& responses) {
+    if constexpr (std::is_same_v<Task, task::regression>) {
```
[the remark this answers is not in the document]
Comment (KulikovNikita) [reply cpp/oneapi/dal/algo/knn/backend/cpu/train_kernel_brute_force.cpp:47]: One of is more suitable if there are many options. In this case std::is_sam_looks more suitable

**Review by KalyanovD — APPROVED** [review]

**Pull request merged**